clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/spring/client/annotation/processor/AbstractCamundaAnnotationProcessor.java

```diff
@@ -17,26 +17,44 @@
 
 import io.camunda.client.CamundaClient;
 import io.camunda.spring.client.bean.ClassInfo;
-import org.springframework.beans.factory.BeanNameAware;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
 
-public abstract class AbstractCamundaAnnotationProcessor implements BeanNameAware {
+public abstract class AbstractCamundaAnnotationProcessor
+    implements ApplicationContextAware, CamundaClientLifecycleAware {
+  private ApplicationContext applicationContext;
 
-  private String beanName;
-
-  public String getBeanName() {
-    return beanName;
+  public ApplicationContext getApplicationContext() {
+    return applicationContext;
   }
 
   @Override
-  public void setBeanName(final String beanName) {
-    this.beanName = beanName;
+  public void setApplicationContext(final ApplicationContext applicationContext) {
+    this.applicationContext = applicationContext;
   }
 
-  public abstract boolean isApplicableFor(ClassInfo beanInfo);
+  protected abstract boolean isApplicableFor(ClassInfo beanInfo);
+
+  protected abstract void configureFor(final ClassInfo beanInfo);
 
-  public abstract void configureFor(final ClassInfo beanInfo);
+  protected abstract void start(CamundaClient client);
 
-  public abstract void start(CamundaClient client);
+  protected abstract void stop(CamundaClient client);
 
-  public abstract void stop(CamundaClient client);
+  @Override
+  public void onStart(final CamundaClient client) {
+    for (final String beanName : getApplicationContext().getBeanDefinitionNames()) {
```

Comment:
```suggestion
    for (final String beanName : applicationContext.getBeanDefinitionNames()) {
```
🔧 you can make use of the field, I guess `getApplicationContext()` is not meant to be overridden by subclasses, is it?